src/SavedSearch.php

```diff
@@ -561,6 +561,9 @@ public function prepareQueryToUse($type, $query_tab, $enable_partial_warnings =
                     unset($query_tab['criteria']);
                     $new_key = 0;
                     foreach ($query_tab_save['criteria'] as $key => $val) {
+                        if (isset($val['itemtype']) && isset($val['meta'])) {
+                            $opt = Search::getCleanedOptions($val['itemtype']);
+                        }
                         if (
```

Comment:
There is a loop on `$query_tab_save['metacriteria']` few line below. Search criteria query string probably changed in the past, and I guess meta criteria where not declared the same way. This code probably handles old query string format, but it contains a check that should probably be added here. Indeed, following extract validates that criterion itemtype is available in meta context of main item.

```php
$meta_ok = Search::getMetaItemtypeAvailable($query_tab['itemtype']);
...
if (!isset($val['itemtype']) || !in_array($val['itemtype'], $meta_ok)) {
    $partial_load = true;
}
```